sdk_contrib/typed_span/src/main/java/io/opentelemetry/sdk/contrib/typedspan/http/HttpClientSpanWrapper.java

```diff
@@ -0,0 +1,31 @@
+package io.opentelemetry.sdk.contrib.typedspan.http;
+
+import static io.opentelemetry.sdk.contrib.typedspan.http.HttpSpanWrapperBuilder.STATUS_CODE_KEY;
+import static io.opentelemetry.sdk.contrib.typedspan.http.HttpSpanWrapperBuilder.STATUS_TEXT_KEY;
+
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Status;
+
+public class HttpClientSpanWrapper extends HttpSpanWrapper {
+  public HttpClientSpanWrapper(Span span) {
+    super(span);
+  }
+
+  public HttpClientSpanWrapper setStatusCode(HttpStatusCode status) {
+    this.getRawSpan().setAttribute(STATUS_CODE_KEY, status.getCode());
+    this.getRawSpan().setAttribute(STATUS_TEXT_KEY, status.getMsg());
+    return this;
+  }
+
+  public HttpClientSpanWrapper setStatusCode(int code) {
+    HttpStatusCode status = new HttpStatusCode(code);
+    this.getRawSpan().setAttribute(STATUS_CODE_KEY, status.getCode());
+    this.getRawSpan().setAttribute(STATUS_TEXT_KEY, status.getMsg());
+    return this;
+  }
+
+  public void end(Status status) {
+    this.getRawSpan().setStatus(status);
+    this.getRawSpan().end();
```

Comment:
```suggestion
    this.getRawSpan().end();
```

Set status already in `setStatusCode`.

Reply:
I agree with your comments, but I suggest to do the opposite:
- remove `setStatus(..)`
- `end(int)` requires the status code and it sets the http status code/text, sets the Span status, and terminate the span.
This way, all HTTP Client data is mandatory

Reply:
Hmm, that's usually OK but it does not allow setting the status multiple times, e.g. when you start out wanting to return 200 but then it turns out that it becomes a 500.

Reply:
I see this use case a good fit for events more than attributes. This way you know the point in time where the two decisions occur. When the span terminates your returning code decision must be unique, if it changes later in time, the span should not be ended in first place.

Reply:
Exactly, it should not be ended in the first place, but it may still be convenient to set the status in the first place.